File: Tier-1/Bin2Dec/bin2dec.py
import sys


def bin2dec():
    if len(sys.argv) < 2:
        print('Enter binary number')
        return

    number = sys.argv[1]

    for n in number:
        if int(n) != 1 and int(n) != 0:
            print('Enter binary number(1s and 0s)')
            return

    current = 1
    decimal = 0
    for n in range(len(number) - 1, -1, -1):
        if n == len(number) - 1:
            if int(number[n]) == 1:
                decimal = decimal + current
            current = current + 1
            continue
        if int(number[n]) == 1:
            decimal = decimal + current
        current = current * 2

    print(decimal)


bin2dec()

File: Tier-1/Bin2Dec/bin2dec.js
function bin2dec(binary) {
  if (process.argv.length < 3) {
    return 'Enter binary number';
  }

  binary = binary.toString();

  for (i = 0; i < binary.length; i++) {
    if (+binary[i] !== 1 && +binary[i] !== 0) {
      return 'Enter binary number(1s and 0s)';
    }
  }

  let current = 1;
  let decimal = 0;
  for (let i = binary.length - 1; i >= 0; i--) {
    if (i === binary.length - 1) {
      if (+binary[i] === 1) decimal += current;
      current++;
      continue;
    }
    if (+binary[i] === 1) decimal += current;
    current *= 2;
  }

  return decimal;
}

const dec = bin2dec(process.argv[2]);
console.log(dec);
